vendor/brave-ios/Ads/BATBraveAds.mm

```diff
@@ -437,7 +437,11 @@ - (void)updateAdRewards:(BOOL)shouldReconcile
 
 - (void)detailsForCurrentCycle:(void (^)(NSInteger adsReceived, double estimatedEarnings, NSDate *nextPaymentDate))completion
 {
-  ads->GetTransactionHistory(^(ads::StatementInfo list) {
+  ads->GetTransactionHistory(^(bool success, ads::StatementInfo list) {
+    if (!success) {
+      return;
```

Comment:
Should call completion before returning here: (`completion(0, 0, nil)`)

Reply:
If not successful ads is not initialized so we do not want to update the UI

Reply:
The UI is already visible at this point, so either we implement an error state or we return with empty data

Reply:
Fixed